src/unix/tty.c

```diff
@@ -61,7 +76,7 @@ int uv_tty_init(uv_loop_t* loop, uv_tty_t* tty, int fd, int readable) {
    * different struct file, hence changing its properties doesn't affect
    * other processes.
    */
-  if (type == UV_TTY) {
+  if (type == UV_TTY && ttystat.st_ino == fdstat.st_ino) {
```

Comment:
The additions above should only be done after it's been established that `type == UV_TTY`.